container/client.go

```diff
@@ -118,17 +119,49 @@ func (client dockerClient) StartContainer(c Container) error {
 	bg := context.Background()
 	config := c.runtimeConfig()
 	hostConfig := c.hostConfig()
+	networkConfig := &network.NetworkingConfig{EndpointsConfig: c.containerInfo.NetworkSettings.Networks}
+	// simpleNetworkConfig is a networkConfig with only 1 network.
+	// see: https://github.com/docker/docker/issues/29265
+	simpleNetworkConfig := func() *network.NetworkingConfig {
+		oneEndpoint := make(map[string]*network.EndpointSettings)
+		for k, v := range networkConfig.EndpointsConfig {
+			oneEndpoint[k] = v
+			// we only need 1
+			break
+		}
+		return &network.NetworkingConfig{EndpointsConfig: oneEndpoint}
+	}()
+
 	name := c.Name()
 
 	log.Infof("Starting %s", name)
-	creation, err := client.api.ContainerCreate(bg, config, hostConfig, nil, name)
+	creation, err := client.api.ContainerCreate(bg, config, hostConfig, simpleNetworkConfig, name)
 	if err != nil {
 		return err
 	}
 
 	log.Debugf("Starting container %s (%s)", name, creation.ID)
 
-	return client.api.ContainerStart(bg, creation.ID, types.ContainerStartOptions{})
+	err = client.api.ContainerStart(bg, creation.ID, types.ContainerStartOptions{})
```

Comment:
Yes, I'm aware of the behaviour that we can't directly create the container with all attached networks. But my question was more if we could do it this way.

1. ContainerCreate(simpleNetworkConfig)
2. NetworkDisconnect(simpleNetworkConfig)
3. NetworkConnect(networkConfig.EndpointsConfig)
4. ContainerStart()

I think starting the container with only one attached network and afterwards disconnect it and reconnecting the correct ones, may result in strange behaviors for the containers.
It seems like [docker-compose](https://github.com/docker/docker/issues/29265#issuecomment-265909198) also first creates the container, then connects/disconnect the networks and as last step starts the container.